x-pack/legacy/plugins/lens/public/indexpattern_plugin/lens_field_icon.tsx

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { ICON_TYPES, palettes } from '@elastic/eui';
+import classNames from 'classnames';
+import { FieldIcon } from '../../../../../../src/plugins/kibana_react/public';
+import { DataType } from '../types';
+
+function stringToNum(s: string) {
```

Comment:
I would like to get ride of this complete logic - the `typeToEuiIconMap` in src/plugins/kibana_react/public/icon/field_icon.tsx should provide the same information so we don't have to do this workaround here anymore with looking up an icon from a field type and then using it as an index for the color palette.

Reply:
This is used in `FieldItem` to determine the color for BarSeries: https://github.com/elastic/kibana/blob/2b03cf7eb41ffab19d157dcd74e49944c95ecdd6/x-pack/legacy/plugins/lens/public/indexpattern_plugin/field_item.tsx#L364

I think leaving the function inside `lens_field_icon` makes sense, but just change it so that it uses `typeToEuiIconMap` internally.